src/main/sarl/fr/utbm/info/vi51/worldswar/environment/Environment.sarl

```diff
@@ -42,4 +68,47 @@ agent Environment {
 		emit(new EnvironmentTerminatedEvent)
 	}
 	
+	// ----------------
+	
+	
+	def computePerceptionGrid : Grid<List<Perceivable>>{
+		var perceptionGrid = new Grid<List<Perceivable>>(0, width-1, 0, height-1)
+		
+		for(var x = 0; x < width; x++) {
+			for(var y = 0; y < height; y++) {
+				// foreach cell
+				
+				var perceptionList : List<Perceivable> = new ArrayList<Perceivable>
+				perceptionGrid.set(x,y, perceptionList)
+				
+				var EnvObjectList = grid.get(x, y).getEnvObjects
+				
+				// foreach Environment object in the cell
+				for(var i:int=0 ; i<EnvObjectList.size ; i++){
```

Comment:
Tu dois pouvoir faire un foreach avec un simple `for(envObj : EnvObjectList)` d'après la doc sarl.